### PR TITLE
Added allocator to dl_util functions.

### PR DESCRIPTION
**Pull request by lundmark**
The dl_util functions were changed to use the internal allocator. The should not be using the internal allocator since that one is dedicated to be used to the type library. Unfortunately there are conditions as to when the allocations are actually done, so we can't use one temp for the file allocations and one for the instance allocations, since they are sometimes the same. This is to solve issue #90.

### CODE REVIEW

File: include/dl/dl_util.h [hunk 1]
```diff
@@ -52,13 +53,15 @@ typedef enum
 		filetype     - Type of file to read, see dl_util_file_type_t.
 		out_instance - Pointer to fill with read instance.
 		out_type     - TypeID of instance found in file, can be set to 0x0.
+		allocator 	 - Allocator for doing temp file allocations
```
Comment (wc-duck) [inline include/dl/dl_util.h]: ... and allocating the out-instance!

File: include/dl/dl_util.h [hunk 2]
```diff
@@ -52,13 +53,15 @@ typedef enum
...
 
 	Returns:
 		DL_ERROR_OK on success.
 */
 dl_error_t dl_util_load_from_file( dl_ctx_t    dl_ctx,       dl_typeid_t         type,
 								   const char* filename,     dl_util_file_type_t filetype,
-								   void**      out_instance, dl_typeid_t*        out_type );
+								   void**      out_instance, dl_typeid_t*        out_type,
+								   dl_allocator *allocator = 0x0 );
```
Comment (wc-duck) [inline include/dl/dl_util.h]: I would like to keep default-values out of the interface as this breaks c-compat.

Comment (wc-duck) [reply include/dl/dl_util.h]: This also need doc that says that 0 is valid an what that means.

Comment (lundmark) [reply include/dl/dl_util.h]: Ait! I will remove default-value and say that 0 is valid!

**Pull request merged**